audit_log/headers.py

```diff
@@ -65,14 +66,14 @@ def parse_spiffe(xfcc_header: str) -> ParsedSPIFFE:
 
 
 def get_principal_from_headers(
-    headers: dict[str, str],
+    headers: typing.Mapping[str, str],
 ) -> Principal:
     """Get principal from headers, supports mTLS, headers set in Istio, and JWTs.
 
     Note: Do not use this to handle your auth, it expects auth to already be handled elsewhere and this is just to help get principals.
 
     Args:
-        headers (dict[str, str]): Headers with all keys lowercase
+        headers (Mapping[str, str]): Headers with all keys lowercase
```

Comment:
as we normalize headers by lowercasing those, we could remove "with all keys lowercase" requirement from doc too? Could be separate PR though

Reply:
Yes that should be changed, the function will recreate a dict with lowercase keys

Reply:
Something like this maybe for the comment?
```suggestion
        headers (Mapping[str, str]): HTTP Headers mapping (note that function will convert all keys to lowercase so case doesn't matter)
```